internal/keepers/service.go

```diff
@@ -161,9 +170,28 @@ func (s *onDemandUpkeepService) start() {
 			for {
 				select {
 				case log := <-ch:
-					// TODO: add delete method to cache; hack sets a new status and 1
-					// nanosecond expire time
-					s.stateCache.Set(string(log.Key), types.Performed, 1)
+					// TODO: check perform log confirmations; expect the log
+					// provider to send a log again at a higher block number
+					// which will have more confirmations
+
+					// update the cache with a performed state. in the case
+					// that this key is queried again, it will not be added to
+					// a report
+					r, cached := s.cache.Get(string(log.Key))
+					if cached {
+						r.State = types.Performed
+						s.cache.Set(string(log.Key), r, defaultExpiration)
+					}
```

Comment:
not sure why we need to update both cache and inFlightCache here. Can separate out the concerns

Reply:
The concept is to block out an upkeep id from being checked. Since the status is isolated to the upkeep result, wherever it is passed, the status should indicate that it should not be eligible. Maybe that's just two statuses then? Eligible and NotEligible?